beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3833,6 +3890,10 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         // end up with blocks in fork choice that are missing from disk.
         // See https://github.com/sigp/lighthouse/issues/2028
         let (_, signed_block, blobs, data_columns) = signed_block.deconstruct();
+        let custody_columns_count = self.data_availability_checker.get_custody_columns_count();
+        let data_columns = data_columns
+            .filter(|a| a.len() >= custody_columns_count)
+            .or_else(|| data_column_recv?.blocking_recv());
```

Comment:
Add a nice comment here explaining the mechanism of `data_column_recv` and why trigger it here under these conditions

Reply:
We should be more explicit here if the builder routine failed. If `blocking_recv` returns None because the sender was dropped we should log an error and track metrics here also. Otherwise it's mixing the signal of data_columns = None because before PeerDAS with data_columns = None on reconstruction error.

Also add a timer histogram around `blocking_recv` to check how much we are blocking this routine.

Reply:
Good catch, If the computation fails for some reasons, we probably want to handle it the same way as "database write failed" and restore fork choice, instead of letting the database commit to a inconsistent state. @michaelsproul thoughts?

Reply:
> Also add a timer histogram around blocking_recv to check how much we are blocking this routine

Metrics added to https://github.com/sigp/lighthouse/pull/6268/commits/278a0b90a97a6443adb4ec173adadac1ac4ec635

Reply:
Addressed error handling in https://github.com/sigp/lighthouse/pull/6268/commits/d51ad5f21cb77be2fafbcf4ed7203a8ae4367906.